src/nvim/os/unix_defs.h

```diff
@@ -2,7 +2,7 @@
 #define NVIM_OS_UNIX_DEFS_H
 
 // Windows doesn't have unistd.h, so we include it here to avoid numerous
-// instances of `#ifdef HAVE_UNISTD_H'.
+// instances of `#ifdef WIN32'.
 #include <unistd.h>
```

Comment:
Using HAVE_UNISTD_H and HAVE_IO_H would be more robust--let's use cmake for what it's good at?

We certainly don't want giant ugly headers, but I suspect these cases will be relatively uncommon. So I don't think we should fret about it.